### PR TITLE
Update parents command to show source mismatch and warn about override

### PR DESCRIPTION
**Pull request by micprog**

### CODE REVIEW

**Review by andreaskurth — COMMENTED** [review]
Could you please add an entry to the Changelog that describes this change? I find it difficult to review this PR and to retrace whether the implementation is doing what it is supposed to do when I don't know what it is supposed to do ;-)

**Comment by micprog** [comment]
Oversight on my part, sorry about that, I added a changelog entry and will squash when merging.

**Review by andreaskurth — COMMENTED** [review]
LGTM, I only have one suggestion to homogenize and deduplicate code

File: src/sess.rs [hunk 1]
```diff
@@ -1370,6 +1370,17 @@ pub enum DependencySource {
     Git(String),
 }
 
+impl<'a> From<&'a config::Dependency> for DependencySource {
+    fn from(cfg: &'a config::Dependency) -> DependencySource {
+        match *cfg {
+            config::Dependency::Path(ref path) => DependencySource::Path(path.clone()),
+            config::Dependency::GitRevision(ref url, _) => DependencySource::Git(url.clone()),
+            config::Dependency::GitVersion(ref url, _) => DependencySource::Git(url.clone()),
+            config::Dependency::Version(_) => DependencySource::Git("not linked".to_string()),
```
Comment (andreaskurth) [inline src/sess.rs:1379]: I think this should be
```suggestion
            config::Dependency::Version(_) => DependencySource::Registry,
```
like here:
https://github.com/pulp-platform/bender/blob/22784177e742d1bfb1543969a1241f6d1bd6587c/src/sess.rs#L133-L136

I suggest to update the `impl From` with the latter code and use the `From` trait where the latter code is currently used.

Comment (andreaskurth) [inline src/sess.rs]: I think this should be
```suggestion
            config::Dependency::Version(_) => DependencySource::Registry,
```
like here:
https://github.com/pulp-platform/bender/blob/22784177e742d1bfb1543969a1241f6d1bd6587c/src/sess.rs#L133-L136

I suggest to update the `impl From` with the latter code and use the `From` trait where the latter code is currently used.

Comment (micprog) [reply src/sess.rs]: Makes sense, I have updated the corresponding sections.

Comment (andreaskurth) [reply src/sess.rs]: Thanks!

**Review by andreaskurth — APPROVED** [review]

**Pull request merged**